src/SourceBuild/content/test/Microsoft.DotNet.UnifiedBuild.Tests/Microsoft.DotNet.UnifiedBuild.Tests.csproj

```diff
@@ -3,48 +3,60 @@
+    <ProjectReference Include="$(RepositoryEngineeringDir)publish.proj" ReferenceOutputAssembly="false" PrivateAssets="all" />
   </ItemGroup>
 
   <ItemGroup>
     <Content Include="assets\**" CopyToOutputDirectory="Always" />
-    <Content Include="xunit.runner.json" CopyToOutputDirectory="PreserveNewest" />
   </ItemGroup>
 
   <Target Name="SetRuntimeConfigOptions"
           DependsOnTargets="DetermineSourceBuiltSdkVersion"
           BeforeTargets="_GenerateRuntimeConfigurationFilesInputCache">
     <ItemGroup>
+      <_ProducedPackages Include="$(ArtifactsShippingPackagesDir)**/*.nupkg" />
+    </ItemGroup>
+    <ItemGroup>
+      <!-- General configs -->
       <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).LogsDirectory">
         <Value>$(ArtifactsTestResultsDir)</Value>
       </RuntimeHostConfigurationOption>
-      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).UbSdkArchivePath">
-        <Value>$(SdkTarballPath)</Value>
-      </RuntimeHostConfigurationOption>
       <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).TargetRid">
         <Value>$(TargetRid)</Value>
       </RuntimeHostConfigurationOption>
       <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).PortableRid">
         <Value>$(PortableRid)</Value>
       </RuntimeHostConfigurationOption>
-      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).UbBuildVersion">
-        <Value>$(SourceBuiltSdkVersion)</Value>
+      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).WarnOnContentDiffs">
+        <Value>$(UBTestsWarnOnContentDiffs)</Value>
       </RuntimeHostConfigurationOption>
-      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).WarnOnSdkContentDiffs">
-        <Value>$(UBTestsWarnOnSdkContentDiffs)</Value>
+      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).AssetManifestPath">
+        <Value>$(ArtifactsDir)VerticalManifest.xml</Value>
+      </RuntimeHostConfigurationOption>
+
+      <!-- SDK archive test configs -->
+      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).UbSdkArchivePath">
+        <Value>$(SdkTarballPath)</Value>
+      </RuntimeHostConfigurationOption>
+      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).UbSdkVersion">
+        <Value>$(SourceBuiltSdkVersion)</Value>
       </RuntimeHostConfigurationOption>
       <!-- Set for local testing to avoid downloading a new SDK each test -->
       <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).MsftSdkArchivePath">
         <Value>$(UBTestsMsftSdkArchivePath)</Value>
       </RuntimeHostConfigurationOption>
-      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).NoDiagnosticMessages">
-        <Value>$(VSTestUseMSBuildOutput)</Value>
+
+      <!-- Nuget package test configs -->
+      <RuntimeHostConfigurationOption Include="$(MSBuildProjectName).Packages">
+        <Value>@(_ProducedPackages->'%(Identity)')</Value>
```

Comment:
I thought the data source is the asset manifest and not the packages?

Reply:
Yeah, the asset manifest still controls the version of the official-built package that the test downloads. This is to get the paths to the VMR-built packages to compare the contents with the downloaded package.